app/plugins/openwhisk.md

```diff
@@ -0,0 +1,255 @@
+closed.Defaults to `60 seconds`.
+
+
+Note: If `config.https_verify` set  as `true` then the server certificate
+will be verified according to the CA certificates specified by the
+`lua_ssl_trusted_certificate` directive in Kong configuration.
+
+----
+
+## Demonstration
+
+For this demonstration we are running Kong and 
+[Openwhisk platform](https://github.com/openwhisk/openwhisk) locally as
+Vagrant machine on MAC.
+
+1. Create a javascript action `hello` with following code
+snippet on Openwhisk platform using [`wsk cli`](https://github.com/openwhisk/openwhisk-cli).
+
+    ```javascript
+    function main(params) {
+        var name = params.name || 'World';
+        return {payload:  'Hello, ' + name + '!'};
+    }
+    ```
+
+    ```bash
+    $ wsk action create hello hello.js
+
+    ok: created action hello
+    ```
+
+2. Create an API on Kong
+
+    ```bash
+    $ curl -i -X  POST http://localhost:8001/apis/ \
+      --data "name=openwhisk-test" -d "hosts=example.com" \
+      --data "upstream_url=http://nowhere.com"
+
+    HTTP/1.1 201 Created
+    ...
+
+    ```
+
+3. Apply the `openwhisk` plugin to the API on Kong
+
+    ```bash
+    $ curl -i -X POST http://localhost:8001/apis/openwhisk-test/plugins \
+        --data "name=openwhisk" \
+        --data "config.host=192.168.33.13" \
+        --data "config.service_token=username:key" \
+        --data "config.action=hello" \
+        --data "config.path=/api/v1/namespaces/guest"
+
+    HTTP/1.1 201 Created
+    ...
+
+    ```
+
+4. Make a request to invoke the action
+
+    **Without parameters**
+
+      ```bash
+      $ curl -i -X POST http://localhost:8000/ -H "Host:hello.com"
+      HTTP/1.1 200 OK
+      ...
+
+      {
+        "payload": "Hello, World!"
+      }
+      ```
+
+    **Parameters as querystring**
+
+      ```bash
+      $ curl -i -X POST http://localhost:8000/?name=foo -H "Host:hello.com"
+      HTTP/1.1 200 OK
+      ...
+
+      {
+        "payload": "Hello, foo!"
+      }
+      ```
+
+    **Parameters as form-urlencoded**
+
+      ```bash
+      $ curl -i -X POST http://localhost:8000/ -H "Host:hello.com" --data "name=bar"
+      HTTP/1.1 200 OK
+      ...
+
+      {
+        "payload": "Hello, bar!"
+      }
+      ```
+
+    **Parameters as JSON body**
+
+      ```bash
+      $ curl -i -X POST http://localhost:8000/ -H "Host:hello.com" \
+        -H "Content-Type:application/json" --data '{"name":"bar"}'
+      HTTP/1.1 200 OK
+      ...
+
+      {
+        "payload": "Hello, bar!"
+      }
+      ```
+
+    **Parameters as multipart form**
+
+      ```bash
+      $ curl -i -X POST http://localhost:8000/ -H "Host:hello.com"  -F name=bar
+      HTTP/1.1 100 Continue
+
+      HTTP/1.1 200 OK
+      ...
+
+      {
+        "payload": "Hello, bar!"
+      }
+      ```
+
+    **OpenWhisk metadata in response**
+
+    By default Kong, `config.result` set as `true`,  will only send the `result`
```

Comment:
I like @theetrain suggestion, I would however remove from his examples *Kong's*, and change `config-result` to `config.result`